src/app/clusters/service-area-server/service-area-server.h

```diff
@@ -0,0 +1,751 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include "service-area-cluster-objects.h"
+#include <app-common/zap-generated/cluster-objects.h>
+
+#include <app/AttributeAccessInterface.h>
+#include <app/CommandHandlerInterface.h>
+#include <app/ConcreteCommandPath.h>
+#include <app/util/af-types.h>
+
+#include <app/util/basic-types.h>
+#include <app/util/config.h>
+#include <lib/support/Span.h>
+#include <platform/CHIPDeviceConfig.h>
+
+#include <list>
+#include <map>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace ServiceArea {
+
+
+// *****************************************************************************
+// cluster constraints
+
+constexpr size_t   kMaxNumSupportedLocations  = 255;
+constexpr size_t   kMaxNumSupportedMaps       = 255;
+constexpr size_t   kMaxNumSelectedLocations   = 255;
+constexpr size_t   kMaxNumSupportedProgress   = 255;
+
+
+/**
+ * Uncopyable is a helper to ensure that Service Area cluster instances and delegates
+ * are unique and cannot be copied.
+ */
+class Uncopyable
```

Comment:
Again, copied from operational-state-server.h.  If common and used for a bunch of declarations of non-copyable classes - pretty cool.
For one or two classes in one file it's more a matter of taste, you could do the same thing inside each class.